SPECS/openssl-pkcs11/openssl-pkcs11.spec

```diff
@@ -1,40 +1,49 @@
+Version: 0.4.12
+Release: 1%{?dist}
+
 # Define the directory where the OpenSSL engines are installed
-%global enginesdir %{_libdir}/engines-1.1
-Summary:        A PKCS#11 engine for use with OpenSSL
+%global enginesdir %{_libdir}/engines-3
+
 Name:           openssl-pkcs11
-Version:        0.4.10
-Release:        10%{?dist}
+Summary:        A PKCS#11 engine for use with OpenSSL
 # The source code is LGPLv2+ except eng_back.c and eng_parse.c which are BSD
-License:        LGPLv2+ AND BSD
+# There are parts licensed with OpenSSL license too
+License:        LGPL-2.1-or-later AND BSD-2-Clause AND OpenSSL
 Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
 URL:            https://github.com/OpenSC/libp11
 Source0:        https://github.com/OpenSC/libp11/releases/download/libp11-%{version}/libp11-%{version}.tar.gz
-Patch0:         openssl-pkcs11-0.4.10-various-bug-fixes.patch
-Patch1:         openssl-pkcs11-0.4.10-search-objects-in-all-matching-tokens.patch
-Patch2:         openssl-pkcs11-0.4.10-add-support-pin-source.patch
-Patch3:         openssl-pkcs11-0.4.10-set-rsa-flag-ext-pkey.patch
+
+# Downstream only for now to make RSA operations working in FIPS mode
 Patch4:         openssl-pkcs11-0.4.10-set-rsa-fips-method-flag.patch
-BuildRequires:  autoconf
-BuildRequires:  automake
-BuildRequires:  libtool
+# unbreak operation when some other engine is present in openssl.cnf
+# https://github.com/OpenSC/libp11/pull/460
+# https://github.com/OpenSC/libp11/commit/feb22a66
+# 580c12b78b63d88010a6178d7c4c58186938c479
+# 74497e0fa5b69b15790d6697e1ebce13af842d4c
+Patch5:         openssl-pkcs11-ossl3.patch
+Patch6:         openssl-pkcs11-ec-copy.patch
+
+BuildRequires: make
+BuildRequires:  autoconf automake libtool
```

Comment:
nit: would prefer these on separate lines

Reply:
agree but I figured I'd keep it this way in the interest of staying as close as reasonably possible to the upstream.